Modules/atexitmodule.c

```diff
@@ -194,33 +181,26 @@ atexit_register(PyObject *module, PyObject *args, PyObject *kwargs)
                 "the first argument must be callable");
         return NULL;
     }
+    PyObject *rest_of_args = PyTuple_GetSlice(args, 1, PyTuple_GET_SIZE(args));
 
-    struct atexit_state *state = get_atexit_state();
-    if (state->ncallbacks >= state->callback_len) {
-        atexit_py_callback **r;
-        state->callback_len += 16;
-        size_t size = sizeof(atexit_py_callback*) * (size_t)state->callback_len;
-        r = (atexit_py_callback**)PyMem_Realloc(state->callbacks, size);
-        if (r == NULL) {
-            return PyErr_NoMemory();
-        }
-        state->callbacks = r;
+    if (kwargs == NULL)
+    {
+        kwargs = Py_None;
     }
-
-    atexit_py_callback *callback = PyMem_Malloc(sizeof(atexit_py_callback));
-    if (callback == NULL) {
-        return PyErr_NoMemory();
+    PyObject *tuple = _PyTuple_FromArray((PyObject *[]){ func, rest_of_args, kwargs }, 3);
```

Comment:
You may rename it to *callback* and use:

```suggestion
    PyObject *callback = PyTuple_Pack(3, func, func_args, func_kwargs);
```